Feature/clion licenses

strahlc opened the pull request:
dev-util/clion: add bindist restriction
dev-util/clion: add licenses for bundled libraries

Package-Manager: Portage-2.3.69, Repoman-2.3.16
Signed-off-by: Christian Strahl <c.a.strahl@gmail.com>

Closes: https://bugs.gentoo.org/694266

mgorny commented:
How did you manage to find all those licenses? Did you just look at all files in licenses dir, or something more?

ulm commented on licenses/CDDL-1.1:
```diff
@@ -0,0 +1,363 @@
+
+     The Covered Software is a "commercial item," as that term is defined in
+     48 C.F.R. 2.101 (Oct. 1995), consisting of "commercial computer software"
+     (as that term is defined at 48 C.F.R. ? 252.227-7014(a)(1)) and
```
Broken encoding? This should be a section sign "§" not a question mark.
Where did you take this license file from?

strahlc replied:
I've rebased this feature branch to the current master, and dropped its CDDL-1.1 license file in favor of the ones in master (from commit https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=9a8505e9128e8e80dc11b330e1b795ca465b9061)

ulm commented on dev-util/clion/clion-2019.2.1.ebuild:
```diff
@@ -9,11 +9,14 @@ DESCRIPTION="A complete toolset for C and C++ development"
+LICENSE="Apache-1.1 Apache-2.0 BSD BSD-2 CC0-1.0 CDDL-1.1 CPL-0.5 CPL-1.0
+	EPL-1.0 EPL-2.0 GPL-2 GPL-2-with-classpath-exception GPL-3 ISC
+	JDOM LGPL-2.1+ LGPL-3 MIT MPL-1.0 MPL-1.1 OFL oromatcher
+	public-domain PSF-2 sun-bcla-javahelp UoI-NCSA ZLIB
```
What files in the package are licensed under sun-bcla-javahelp?
Javahelp used to be licensed under GPL-2-with-classpath-exception, and was recently relicensed to CDDL-1.1.

strahlc replied:
you're right, my fault.

ulm replied:
oromatcher is gone as well? Was that for lib/oro-2.0.8.jar (which really is Apache-1.1)?

strahlc commented on dev-util/clion/clion-2019.2.1.ebuild:
```diff
@@ -9,11 +9,14 @@ DESCRIPTION="A complete toolset for C and C++ development"
+	EPL-1.0 EPL-2.0 GPL-2 GPL-2-with-classpath-exception GPL-3 ISC
+	JDOM LGPL-2.1+ LGPL-3 MIT MPL-1.0 MPL-1.1 OFL oromatcher
+	public-domain PSF-2 sun-bcla-javahelp UoI-NCSA ZLIB
+	|| ( IDEA IDEA_Academic IDEA_Classroom IDEA_OpenSource IDEA_Personal )"
```
you're right, I've reordered the licenses

strahlc commented:
They've also removed the `oromatcher.jar`, but they still deliver `/opt/clion-2019.2.1/license/oromatcher_license.txt`.
Due to that I've removed oromatcher from the `LICENSE` list and removed the license file as well.
